course-material/advanced-querying-techniques/subqueries.md

```diff
@@ -1,14 +1,316 @@
+# Naming our folder that will hold our .csv files
+output = "expanded_data"
+banking_data_script.extract_asc_to_csv(link, output)
+```
+
+If you ran the above cell, you should have a folder `expanded_data` in your current directory that contains the `.csv` files we will be using. In this tutorial, we will be focusing on three of these files: `loan.csv`, `account.csv`, `district.csv`.
+
+## Load Engine
+
+We now load in our SQL extension that allows us to execute SQL queries in Jupyter Notebooks.
+
+```{important}
+<b>Note</b> Ensure you restart any previous notebook that has the same database name as the one initialized below.
+```
+
+```{code-cell} ipython3
+# Loading in SQL extension
+%reload_ext sql
+# Initiating a DuckDB database named 'bank_data.duck.db' to run SQL queries
+%sql duckdb:///bank_data.duck.db
+```
+
+<!-- endregion -->
+
+## Creating Tables
+
+Let's start off with loading three of the eight `.csv` files from the `expanded_data` folder in the current directory to our newly created DuckDB database. Like in the previous tutorial, we will [create a schema](https://ploomber-sql.readthedocs.io/en/latest/intro-to-sql/joining-data-in-sql.html#creating-a-schema) `s1` in which we will store the tables. Here we use the `CREATE TABLE` syntax in DuckDB to ingest four of the eight `.csv` files. The `read_csv_auto` is a function that helps SQL understand our local `.csv` file for creation into our database.
+
+
+```{code-cell} ipython3
+%%sql
+CREATE SCHEMA s1;
+CREATE TABLE s1.account AS
+FROM read_csv_auto('expanded_data/account.csv', header=True, sep=',');
+CREATE TABLE s1.district AS
+FROM read_csv_auto('expanded_data/district.csv', header=True, sep=',');
+CREATE TABLE s1.loan AS
+FROM read_csv_auto('expanded_data/loan.csv', header=True, sep=',');
+```
+
+## Queries
+
+Let's first load each table for reference and to better understand their contents.
+
+```{code-cell} ipython3
+%%sql
+SELECT *
+FROM s1.account
+LIMIT 5
+```
+
+```{code-cell} ipython3
+%%sql
+SELECT *
+FROM s1.district
+LIMIT 5
+```
+
+```{code-cell} ipython3
+%%sql
+SELECT *
+FROM s1.loan
+LIMIT 5
+```
+
+### Subquery in SELECT
+
+The query belows results in the average loan amount for each `account_id` by having a subquery in the `SELECT` clause.
+
+```{code-cell} ipython3
+%%sql
+SELECT s1.account.account_id, s1.account.date, 
+      (SELECT AVG(s1.loan.amount) 
+      FROM s1.loan 
+      WHERE s1.loan.account_id = s1.account.account_id
+      ) AS average_loan_amount
+FROM s1.account 
+WHERE average_loan_amount IS NOT NULL
+ORDER BY average_loan_amount
+LIMIT 5
+```
+
+The above SQL statement is quite verbose. Let's instead utilize aliases to make the query more readable.
+
+```{code-cell} ipython3
+%%sql
+SELECT a.account_id,
+      (SELECT AVG(l.amount) 
+      FROM s1.loan AS l 
+      WHERE l.account_id = a.account_id
+      ) AS average_loan_amount
+FROM s1.account AS a
+WHERE average_loan_amount IS NOT NULL
+ORDER BY average_loan_amount
+```
+
```

Comment:
After you introduce the 2 queries, now show how you can easily handle large queries with JupySQL with the --save option

https://jupysql.ploomber.io/en/latest/compose.html